### PR TITLE
Move quirks initialization to the hid_pidff_init function

### PR DESCRIPTION
**Pull request by JacKeTUs**

### CODE REVIEW

**Comment by JacKeTUs** [comment]
@Lawstorant, please check if Moza wheelbase initializes for you

File: hid-pidff.h [hunk 1]
```diff
@@ -20,7 +20,6 @@
 #define PIDFF_QUIRK_NO_DELAY_EFFECT         BIT(2)
 
 
-int hid_pidff_init(struct hid_device *hid);
-int hid_pidff_init_with_quirks(struct hid_device *hid, const struct hid_device_id *id);
+int hid_pidff_init(struct hid_device *hid, const struct hid_device_id *id);
```
Comment (Lawstorant) [inline hid-pidff.h:23]: This conflicts with in-kernel pidff init function. Maybe rename it to `hid_new_pidff_init`?

File: hid-pidff.c [hunk 2]
```diff
@@ -1320,9 +1320,9 @@ static int pidff_check_autocenter(struct pidff_device *pidff,
 }
 
 /*
- * Check if the device is PID and initialize it
+ * Check if the device is PID and initialize it, with quirks
  */
-int hid_pidff_init(struct hid_device *hid)
+int hid_pidff_init(struct hid_device *hid, const struct hid_device_id *id)
```
Comment (Lawstorant) [inline hid-pidff.c:1325]: Just a pointer to function definition

File: hid-pidff-wrapper.c [hunk 3]
```diff
@@ -69,9 +69,9 @@ static int universal_pidff_probe(struct hid_device *hdev,
 		goto err;
 	}
 
-	ret = hid_pidff_init_with_quirks(hdev, id);
+	ret = hid_pidff_init(hdev, id);
```
Comment (Lawstorant) [inline hid-pidff-wrapper.c:72]: Pointer to function reference

**Review by Lawstorant — CHANGES_REQUESTED** [review]

**Comment by Lawstorant** [comment]
Apart from this one issue, it compiles successfully after changing the name of this function and initializes MOZA FFB properly.

File: hid-pidff.h (hunk 1 above)
Comment (JacKeTUs) [reply hid-pidff.h:23]: yep, let's rename it for now
Now pidff driver has hardcoded descriptor values and we can't set any of specific descriptors as non-mandatory, that's why we need the quirks. 

ff_init function from driver (`include/linux/hid.h:1213`) does not expect any quirks from pidff, it expects USB quirks, a different type

Comment (Lawstorant) [reply hid-pidff.h]: Yup, I give up on upstream pidff compatibility for now. Apart from pushing the 0 length issue upstream, I don't think we will achieve much more there. At least this give us more freedom

**Review by Lawstorant — APPROVED** [review]
LGTM

**Pull request merged**